meta-lmp-base/classes/lmp.bbclass

```diff
@@ -20,6 +20,22 @@ sota_fstab_update() {
 
 ROOTFS_POSTPROCESS_COMMAND:append:sota = " sota_fstab_update; "
 
+preload_apps () {
+        if [ -f "${APP_PRELOADER}" ] && [ -r "${APP_PRELOADER}" ] && [ -x "${APP_PRELOADER}" ]; then
```

Comment:
probably yes, the `-x` has the  `-f` implied

```
-r
file has read permission (for the user running the test)
```
^ `bitbake` building the image inside the `pseudo` that is like a `fakeroot` so `-r` is not needed

Reply:
`-x`  returns `true` for a folder, so to make sure that it is a file you have to use `-f`.
Also, `-r` is used to make sure that a file is readable by a given user, because both `-x` and `-f` can return `true` while the file cannot be executed because a current user doesn't have a read permission :). https://stackoverflow.com/questions/10319652/check-if-a-file-is-executable

Reply:
The `exec` access is enough for binary files, but for bashed scripts, a `read` right is required.
```
$ ls -l z
---x--x--x 1 cryo cryo 24 тра 27 19:41 z
$ [ -f z ] && [ -x z ] && echo OK
OK
$ ./z 
/bin/bash: ./z: Permission denied
```$ ./z.sh 
/usr/bin/python: can't open file '/home/cryo/tmp/./z.sh': [Errno 13] Permission denied

The same for python:
```
$ ./z.sh 
/usr/bin/python: can't open file '/home/cryo/tmp/./z.sh': [Errno 13] Permission denied
```

Are we sure the APP_PRELOADER will always be binary?